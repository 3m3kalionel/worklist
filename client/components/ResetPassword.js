import React from 'react';
import { Mutation } from 'react-apollo';

import AuthFormStyles from '../styles/AuthFormStyles';
import { WithForm } from './HOC';

import RESET_PASSWORD_MUTATION from '../graphql/mutations/ResetPassword';
import CURRENT_USER_QUERY from '../graphql/queries/CurrentUser';

const WrappedComponent = props => {
  const {
    onChange,
    onSubmit,
    onError,
    values,
    resetToken
  } = props;
  const { password, confirmPassword } = values;
  
  return (
    <Mutation
      mutation={RESET_PASSWORD_MUTATION}
      variables={{ resetToken, password, confirmPassword }}
      refetchQueries={[{ query: CURRENT_USER_QUERY }]}
    >
      {(resetPassword, { data, error, loading }) => (
        <form
          onSubmit={onSubmit(resetPassword, 'reset')}
        >
          <AuthFormStyles>
            <div>
              <div id="authentication-form">
                <div id="form-body">
                  {error && onError(error)}
                  {props.children}
                  <h1 id="form-header">Reset Password</h1>
                    <label htmlFor="password">Password</label>
                    <input
                      name="password"
                      type="password"
                      className="auth-form-input"
                      placeholder="Password"
                      value={password}
                      autoComplete="current-password"
                      onChange={onChange}
                      required
                    />
                    <label htmlFor="confirmPassword">Confirm Password</label>
                    <input
                      name="confirmPassword"
                      type="password"
                      className="auth-form-input"
                      placeholder="Confirm Password"
                      value={confirmPassword}
                      autoComplete="new-password"
                      onChange={onChange}
                      required
                    />
                  <div id="action-button-container">
                    <button type="submit" id="submit-button">Reset Password</button>
                  </div>
                </div>
              </div>
            </div>
          </AuthFormStyles>
        </form>
      )}
    </Mutation>
  )
}

const ResetPassword = WithForm(WrappedComponent);

export default (props) => (
  <div>
    <ResetPassword
      resetToken={props.resetToken}
      initialState={{ password: "", confirmPassword: "" }}
    />
  </div>
);
