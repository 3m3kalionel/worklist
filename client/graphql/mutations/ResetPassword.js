import gql from 'graphql-tag';

const RESET_PASSWORD_MUTATION = gql`
  mutation RESET_PASSWORD_MUTATION(
    $resetToken: String!,
    $password: String!,
    $confirmPassword: String!
  ) {
    resetPassword(
      resetToken: $resetToken,
      password: $password,
      confirmPassword: $confirmPassword
    ) {
      id
      email
    }
  }
`;

export default RESET_PASSWORD_MUTATION;
