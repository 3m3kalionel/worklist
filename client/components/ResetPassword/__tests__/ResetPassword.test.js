import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ApolloProvider } from 'react-apollo';
import '@testing-library/react/cleanup-after-each';
import { createMockClient } from 'mock-apollo-client';

import ResetPassword from '../ResetPassword';

import RESET_PASSWORD_MUTATION from '../../../graphql/mutations/ResetPassword';

jest.mock("../../HOC.js", () => Component => {
  const mockHOC = props => (
    <div><Component {...props} /></div>
  );

   return mockHOC;
});

const setup = (values, onSubmit) => {
  const onSubmitHandler = onSubmit || jest.fn();
  const mockClient = createMockClient();

  mockClient.setRequestHandler(
    RESET_PASSWORD_MUTATION,
    () => Promise.resolve({
      data: {
        resetPassword: {
          id: "cjzdoavvjvgby0b53mie0fqc3",
          username: "gee",
          email: "gee@gmail.com",
          profilePicUrl: null,
          authoredTodoLists: [],
          todoListsContributedTo: []
        }
      }
    })
  );

  const {
    container,
    getByText,
    getByTestId
  } = render(
    <ApolloProvider client={mockClient}>
      <ResetPassword
        values={values}
        onSubmit={onSubmitHandler}
        onChange={jest.fn()}
      />
    </ApolloProvider>
  );

  return { container, getByText, getByTestId }
}

describe('<Signup />', () => {
  test('renders an empty form', () => {
    const { container } = setup({
      password: "",
      confirmPassword: "",
    });

    expect(container).toMatchSnapshot();
  });

  test('renders text in the form input', () => {
    const { container } = setup({
      password: "password",
      confirmPassword: "confirmPassword",
    });

    expect(container).toMatchSnapshot();
  });

  test('calls the submit handler when button is clicked', () => {
    const onSubmitHandler = jest.fn();
  
      const {
        container,
        getByTestId
      } = setup({
          password: "password",
          confirmPassword: "confirmPassword",
        },
        onSubmitHandler
      );
      const form = getByTestId('form');
      fireEvent.submit(form);
  
      expect(onSubmitHandler).toHaveBeenCalled();
      expect(container).toMatchSnapshot();
    });
})
