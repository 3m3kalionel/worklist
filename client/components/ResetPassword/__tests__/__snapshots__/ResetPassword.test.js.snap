// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Signup /> calls the submit handler when button is clicked 1`] = `
<div>
  <div>
    <form
      data-testid="form"
    >
      <div
        class="AuthFormStyles-js87jd-0 ieHHDD"
      >
        <div>
          <div
            id="authentication-form"
          >
            <div
              id="form-body"
            >
              <h1
                id="form-header"
              >
                Reset Password
              </h1>
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="auth-form-input"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value="password"
              />
              <label
                for="confirmPassword"
              >
                Confirm Password
              </label>
              <input
                autocomplete="new-password"
                class="auth-form-input"
                name="confirmPassword"
                placeholder="Confirm Password"
                required=""
                type="password"
                value="confirmPassword"
              />
              <div
                id="action-button-container"
              >
                <button
                  id="submit-button"
                  type="submit"
                >
                  Reset Password
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Signup /> renders an empty form 1`] = `
<div>
  <div>
    <form
      data-testid="form"
    >
      <div
        class="AuthFormStyles-js87jd-0 ieHHDD"
      >
        <div>
          <div
            id="authentication-form"
          >
            <div
              id="form-body"
            >
              <h1
                id="form-header"
              >
                Reset Password
              </h1>
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="auth-form-input"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value=""
              />
              <label
                for="confirmPassword"
              >
                Confirm Password
              </label>
              <input
                autocomplete="new-password"
                class="auth-form-input"
                name="confirmPassword"
                placeholder="Confirm Password"
                required=""
                type="password"
                value=""
              />
              <div
                id="action-button-container"
              >
                <button
                  id="submit-button"
                  type="submit"
                >
                  Reset Password
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`<Signup /> renders text in the form input 1`] = `
<div>
  <div>
    <form
      data-testid="form"
    >
      <div
        class="AuthFormStyles-js87jd-0 ieHHDD"
      >
        <div>
          <div
            id="authentication-form"
          >
            <div
              id="form-body"
            >
              <h1
                id="form-header"
              >
                Reset Password
              </h1>
              <label
                for="password"
              >
                Password
              </label>
              <input
                autocomplete="current-password"
                class="auth-form-input"
                name="password"
                placeholder="Password"
                required=""
                type="password"
                value="password"
              />
              <label
                for="confirmPassword"
              >
                Confirm Password
              </label>
              <input
                autocomplete="new-password"
                class="auth-form-input"
                name="confirmPassword"
                placeholder="Confirm Password"
                required=""
                type="password"
                value="confirmPassword"
              />
              <div
                id="action-button-container"
              >
                <button
                  id="submit-button"
                  type="submit"
                >
                  Reset Password
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
  </div>
</div>
`;
